template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hive/HiveMetastoreCloudStorageServiceConfigProvider.java

```diff
@@ -0,0 +1,54 @@
+package com.sequenceiq.cloudbreak.cmtemplate.configproviders.hive;
+
+import static com.sequenceiq.cloudbreak.cmtemplate.configproviders.ConfigUtils.config;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.stereotype.Component;
+
+import com.cloudera.api.swagger.model.ApiClusterTemplateConfig;
+import com.sequenceiq.cloudbreak.cmtemplate.CmTemplateComponentConfigProvider;
+import com.sequenceiq.cloudbreak.cmtemplate.CmTemplateProcessor;
+import com.sequenceiq.cloudbreak.cmtemplate.configproviders.ConfigUtils;
+import com.sequenceiq.cloudbreak.template.TemplatePreparationObject;
+
+@Component
+public class HiveMetastoreCloudStorageServiceConfigProvider implements CmTemplateComponentConfigProvider {
+
+    private static final String HMS_METASTORE_DIR = "hive.metastore.warehouse.dir";
+
+    private static final String HMS_METASTORE_EXTERNAL_DIR = "hive.metastore.warehouse.external.dir";
+
+    private static final String HMS_METASTORE_DIR_TEMPLATE_PARAM = "hive_warehouse_directory";
+
+    private static final String HMS_METASTORE_EXTERNAL_DIR_TEMPLATE_PARAM = "hive_warehouse_external_directory";
+
+    @Override
+    public List<ApiClusterTemplateConfig> getServiceConfigs(CmTemplateProcessor templateProcessor, TemplatePreparationObject source) {
+        List<ApiClusterTemplateConfig> hmsConfigs = new ArrayList<>(2);
+        ConfigUtils.getStorageLocationForServiceProperty(source, HMS_METASTORE_DIR)
+                .map(location -> hmsConfigs.add(config(HMS_METASTORE_DIR_TEMPLATE_PARAM, location.getValue())));
```

Comment:
Not a big deal™ but `map` creates a new `Optional` which is then discarded.  I think the original `ifPresent` as written in `HiveMetastoreCloudStorageRoleConfigProvider ` was both cheaper and more clear.

Reply:
Changed to ifPresent